packages/common/src/utils/secrets.ts

```diff
@@ -0,0 +1,56 @@
+import { NetworkVMType } from '@avalabs/vm-module-types';
+
+export const getLegacyXPAddressIndexFromPath = (path: string) => {
+  const unprefixed = path.replace('m/', '');
+  const [, , , , addressIndex] = unprefixed.split('/');
```

Comment:
I'm just thinking about my dumb self, so would it make sense to check the length of the unprefixed split string's array's length? 🤔